src/Altinn.App.Api/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -95,6 +98,30 @@ IWebHostEnvironment env
         services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
     }
 
+    /// <summary>
+    /// <para>
+    /// Configures the <see cref="MaskinportenClient"/> service with a configuration object which will be static for the lifetime of the service.
+    /// </para>
+    /// <para>
+    /// If you have already provided a <see cref="MaskinportenSettings"/> configuration, either manually or
+    /// implicitly via <see cref="WebHostBuilderExtensions.ConfigureAppWebHost"/>, this will be overridden.
+    /// </para>
+    /// </summary>
+    /// <param name="services">The service collection</param>
+    /// <param name="configureOptions">
+    /// Action delegate that provides <see cref="MaskinportenSettings"/> configuration for the <see cref="MaskinportenClient"/> service
+    /// </param>
+    public static IServiceCollection ConfigureMaskinportenClient(
+        this IServiceCollection services,
+        Action<MaskinportenSettings> configureOptions
+    )
+    {
+        services.RemoveOptions<MaskinportenSettings>();
+        services.AddOptions<MaskinportenSettings>().Configure(configureOptions).ValidateDataAnnotations();
```

Comment:
In our case, I'm not sure if that's possible due to the `[Required]` attributes and how we call `ValidateDataAnnotations()` after loading the config.

This was because we wanted the JSON to be validated... but if I'm not remembering wrong, this means we have to provide a valid object each time if manually configuring.

That said, I'd be happy to remove the `RemoveOptions` call if that goes against common convention (of which I know very little, unfortunately).

Reply:
I think the validations run after all configures/postconfigures are ran: https://github.com/dotnet/runtime/blob/c1cebefa8b0522a120a455d0b768fc297134523d/src/libraries/Microsoft.Extensions.Options/src/OptionsFactory.cs#L55-L92

Reply:
I've removed the `RemoveOptions` call, but opted to leave the `public required T ...` in the model, since this makes the downstream code a bit cleaner. So in practice you couldn't initialize a partial configuration, then later come back and finish the job. 

I have, however, validated that the last value added overrides any previous values. Eg. the unit test for this covers adding two configs and validating that the last config added is the one retrieved with `IOptionsMonitor.CurrentValue`.